lib/group_domain_models/lib/src/models/school_class.dart

```diff
@@ -142,8 +139,10 @@ class SchoolClassData {
     );
   }
 
-  factory SchoolClassData.fromData(
-      {@required String id, @required Map<String, dynamic> data}) {
+  factory SchoolClassData.fromData({
+    required String id,
+    required Map<String, dynamic> data,
```

Comment:
Nit (maybe for some other PR): For some classes `data` is positional and `id` is named and for example here both is named. We should decide on one of both for all cases.